### DISCUSSION TITLE
Disable ctrl+d with specific window title

### DISCUSSION THREAD

**Discussion by miranyx** [fish-shell/fish-shell#9581] [Q&A] [answered]
Hello,

I would like to disable ctrl+d but only for one specific shell in a terminal which has a custom window title could I somehow disable it for just that shell?

**Comment by faho** [comment, marked as answer]
Fish doesn't know the window title and the escape sequence to get it from the terminal is, from what I can find, not well-supported because of security concerns (see e.g. [VTE's code](https://gitlab.gnome.org/GNOME/vte/-/blob/abb7afd06c64769ff8754d2d82edd660cee16c64/src/vteseq.cc#L9219-9230), which sends a fake static title).

I would recommend you find some other way to identify that window, like set your terminal to pass an environment variable.

For example, if a variable NO_EXIT was set:

```fish
bind \cd 'if set -q NO_EXIT; commandline -f delete-char; else; commandline -f delete-or-exit; end'
```

Or just set ctrl-d to delete-char and use `exit` manually:

```fish
bind \cd delete-char
```

**Comment by lengyijun** [comment]
What's your terminal emulator ?

**Reply by miranyx** [reply]
My terminal emulator is kitty
